### PR TITLE
Feature/2.1 upgrade

### PR DESCRIPTION
**Pull request by emoralesb05**
## Description

Upgrade @angular dependencies
Fixes https://github.com/Teradata/covalent/issues/141

### What's included?

- Upgrade `angular` dependencies to `@2.2.1`
- Upgrade `zone.js` to `@0.6.25`
- Upgrade `angular-cli` to `beta.19-3`
- Upgrade `material` to `alpha.10` and fixed breaking changes https://github.com/angular/material2/blob/master/CHANGELOG.md#200-alpha10-mithril-hoverboard-2016-11-10
- Fix bug for https://github.com/angular/angular/issues/12583

#### Test Steps

- [x] `npm uninstall -g angular-cli`
- [x] `npm cache clean`
- [x] `npm install -g angular-cli@latest`
- [x] `rm -rf node_modules dist tmp`
- [x] `npm i`
- [x] `ng serve`
- [x] See docs still working

### CODE REVIEW

File: src/app/components/components/data-table/data-table.component.html [hunk 1]
```diff
@@ -19,84 +44,55 @@ <h4 class="md-subhead">with custom headings, columns, and inline editing</h4>
               <th td-data-table-column 
                   *ngFor="let column of columns"
                   [numeric]="column.numeric">
-                {{column.label}}
+                { {column.label}}
```
Comment (ilsiepotamus) [inline src/app/components/components/data-table/data-table.component.html]: formatting?

File: src/app/components/components/data-table/data-table.component.html [hunk 2]
```diff
@@ -19,84 +44,55 @@ <h4 class="md-subhead">with custom headings, columns, and inline editing</h4>
...
               </th>
               <tr td-data-table-row *ngFor="let row of basicData">
                 <td td-data-table-cell (click)="openPrompt(row, 'comments')">
-                  <button md-button [class.md-accent]="!row['comments']">{{row['comments'] || 'Add Comment'}}</button>
+                  <button md-button [class.md-accent]="!row['comments']">{ {row['comments'] || 'Add Comment'}}</button>
```
Comment (ilsiepotamus) [inline src/app/components/components/data-table/data-table.component.html]: here also

File: src/app/components/components/data-table/data-table.component.html [hunk 3]
```diff
@@ -19,84 +44,55 @@ <h4 class="md-subhead">with custom headings, columns, and inline editing</h4>
...
                 </td>
                 <td td-data-table-cell *ngFor="let column of columns" [numeric]="column.numeric">
-                  {{row[column.name]}}
+                  { {row[column.name]}}
```
Comment (ilsiepotamus) [inline src/app/components/components/data-table/data-table.component.html]: and here

File: src/app/components/components/data-table/data-table.component.html (hunk 1 above)
Comment (emoralesb05) [inline src/app/components/components/data-table/data-table.component.html]: Its an angular2 issue where if it finds `{{` anywhere in an html file, it passes it to its compiler. 

So we need to add a space between them to be able to render it in the highlight component.

File: src/app/components/components/data-table/data-table.component.html (hunk 2 above)
Comment (kyleledbetter) [inline src/app/components/components/data-table/data-table.component.html]: yup good catch but like Ed said we have to do this to render code snippets in highlight.js for docs

**Pull request merged**